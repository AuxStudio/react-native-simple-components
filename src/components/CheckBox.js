import React from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import styleConstants from "../assets/styleConstants";

import Touchable from "./Touchable";
import { AnimateScale } from "react-native-simple-animators";
import Icon from "react-native-vector-icons/MaterialIcons";

export default function CheckBox(props) {
    // static get propTypes() {
    //     return {
    //         isChecked: PropTypes.bool,
    //         handlePress: PropTypes.func,

    //         // style: PropTypes.node,
    //         // iconStyle: PropTypes.node,
    //     };
    // }

    const icon = props.isChecked && (
        <AnimateScale initialValue={0.5} finalValue={1} shouldAnimateIn>
            <Icon name="check" style={[styles.icon, props.iconStyle]} />
        </AnimateScale>
    );

    return (
        <Touchable
            onPress={props.handlePress}
            disableFeedback
            style={[styles.container, props.style]}>
            {icon}
        </Touchable>
    );
}

const styles = StyleSheet.create({
    container: {
        borderRadius: 8,
        overflow: "hidden",
        borderWidth: 1,
        borderColor: styleConstants.primaryText,
        width: 32,
        height: 32,
        justifyContent: "center",
        alignItems: "center",
    },
    icon: {
        fontSize: styleConstants.iconFont,
        color: styleConstants.primaryText,
    },
});
