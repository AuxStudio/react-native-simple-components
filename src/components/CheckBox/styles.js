import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    overflow: 'hidden',
    borderWidth: 1,
    borderColor: styleConstants.primaryText,
    width: 32,
    height: 32,
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    fontSize: styleConstants.iconFont,
    color: styleConstants.primaryText,
  },
});

export default styles;
