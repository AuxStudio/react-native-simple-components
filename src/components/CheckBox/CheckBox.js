import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes, Text } from 'react-native';
import { AnimateScale } from 'react-native-simple-animators';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';
import Touchable from '../Touchable';

const propTypes = {
  isChecked: PropTypes.bool,
  handlePress: PropTypes.func,
  customIcon: PropTypes.node,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {};

const CheckBox = ({ isChecked, handlePress, customIcon, iconStyle, style }) => {
  const iconComponent = customIcon || <Icon name="check" style={[styles.icon, iconStyle]} />;

  const checkBoxComponent = isChecked && (
    <AnimateScale initialValue={0} finalValue={1} shouldAnimateIn>
      {iconComponent}
    </AnimateScale>
  );

  return (
    <Touchable onPress={handlePress} disableFeedback style={[styles.container, style]}>
      {checkBoxComponent}
    </Touchable>
  );
};

CheckBox.propTypes = propTypes;
CheckBox.defaultProps = defaultProps;

export default CheckBox;
