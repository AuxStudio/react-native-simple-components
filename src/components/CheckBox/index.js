import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes, Text, Platform } from 'react-native';
import Animator from 'react-native-simple-animators';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';
import Touchable from '../Touchable';

const propTypes = {
  isChecked: PropTypes.bool,
  handlePress: PropTypes.func,
  customIcon: PropTypes.node,
  androidRipple: PropTypes.bool,
  androidRippleColor: PropTypes.string,
  androidRippleBorderless: PropTypes.bool,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {};

const CheckBox = ({
  isChecked,
  handlePress,
  customIcon,
  androidRipple,
  androidRippleColor,
  androidRippleBorderless,
  iconStyle,
  style,
}) => {
  const disableFeedback = Platform.OS === 'ios' || !androidRipple;

  const iconComponent = customIcon || <Icon name="check" style={[styles.icon, iconStyle]} />;

  const checkBoxComponent = isChecked && (
    <Animator type="scale" initialValue={0} finalValue={1} shouldAnimateIn>
      {iconComponent}
    </Animator>
  );

  return (
    <Touchable
      onPress={handlePress}
      disableFeedback={disableFeedback}
      style={[styles.container, style]}
      androidRipple={androidRipple}
      androidRippleColor={androidRippleColor}
      androidRippleBorderless={androidRippleBorderless}
    >
      {checkBoxComponent}
    </Touchable>
  );
};

CheckBox.propTypes = propTypes;
CheckBox.defaultProps = defaultProps;

export default CheckBox;
