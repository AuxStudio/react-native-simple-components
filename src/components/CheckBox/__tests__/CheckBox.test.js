import React from 'react';
import renderer from 'react-test-renderer';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

jest.mock('react-native-simple-animators', () => 'Animator');

import CheckBox from '../'; // eslint-disable-line

describe('CheckBox', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <CheckBox
          isChecked
          handlePress={jest.fn()}
          androidRipple
          androidRippleColor="red"
          androidRippleBorderless
          iconStyle={{ color: 'red' }}
          style={{ backgroundColor: 'blue' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<CheckBox />)).toMatchSnapshot();
  });

  it('renders with a customIcon', () => {
    expect(
      renderer.create(<CheckBox customIcon={<MaterialIcon name="check" />} />),
    ).toMatchSnapshot();
  });
});
