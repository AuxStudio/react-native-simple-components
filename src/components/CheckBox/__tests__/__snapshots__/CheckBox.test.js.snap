// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders with a customIcon 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#212121",
        "borderWidth": 1,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      undefined,
    ]
  }
/>
`;

exports[`CheckBox renders with all props 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#212121",
        "borderWidth": 1,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <Animator
    finalValue={1}
    initialValue={0}
    shouldAnimateIn={true}
    type="scale"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#212121",
              "fontSize": 24,
            },
            Object {
              "color": "red",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î—Š
    </Text>
  </Animator>
</View>
`;

exports[`CheckBox renders with minimum required props 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#212121",
        "borderWidth": 1,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      undefined,
    ]
  }
/>
`;
